File: backend/src/main/java/com/newbyproof/rolecall/entity/Video.java
package com.newbyproof.rolecall.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Video {
    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne
    private User trainee;

    private String title;
    private String storageKey; // For s3
    private String contentType;

    @Enumerated(EnumType.STRING)
    private Status status; // PENDING, UPLOADED, REVIEWED

    private Instant createdAt;
}

File: frontend/src/types/Video.ts
export interface Video {
    id: string;
    title: string;
    storageKey: string;
    contentType: string;
    createdAt: string;
    status: "PENDING" | "UPLOADED" | "REVIEWED";
}

export interface CreateUploadUrlRequest {
    title: string;
    filename: string;
    contentType: string;
}

export interface CreateUploadUrlResponse {
    videoId: string;
    uploadUrl: string;
}

export interface DownloadUrlResponse {
    url: string;
}
